src/resources/common.ts

```diff
@@ -98,6 +98,10 @@ export const make = <T extends Resource<string>>(
       cache[typename][name].registerPromise = cache[typename][name][
         'register'
       ]();
+    } else {
+      throw new Error(
+        `Resources must have a unique identifier - ${name} of type ${typename}.\r\nE.g. resources.topic('test') and resources.queue('test') conflict.`
+      );
```

Comment:
Are we sure we want this? I know we've discussed this in the past, but wanted to clarify and agree on a position.

I noted that the current code uses `typeof` as the typename as well, this will only work for javascript types right? Wouldn't we want to use the prototype name in this case?

Reply:
Removed the throw, and corrected the underlying issue as suggested.